lib/screens/ge_repo.dart

```diff
@@ -49,6 +53,13 @@ class GeRepoScreen extends StatelessWidget {
             await auth.fetchGitee('/repos/$owner/$name/branches').then((v) {
           return [for (var branch in v) GiteeBranch.fromJson(branch)];
         });
+        isStarred = await auth
+            .fetchGitee('/user/starred/$owner/$name', requestType: 'NO CONTENT')
+            .then((v) => v.statusCode == HttpStatus.noContent);
+        isWatching = await auth
+            .fetchGitee('/user/subscriptions/$owner/$name',
+                requestType: 'NO CONTENT')
+            .then((v) => v.statusCode == HttpStatus.noContent);
```

Comment:
Could these two variables be included in the returned tuple?

Reply:
No. I tried that first. The MutationButton does not refresh when setState is called if I pass them in the tuple.

Reply:
In this case, these two variables would be redeclared on every build

Perhaps what you want is a `StatefulWidget`?

Reply:
BTW, I got the impression that the second argument of `bodyBuilder` here should work in this case:

https://github.com/git-touch/git-touch/blob/780970a5b754c7d7ceffb6f85a1b61383bbccf17/lib/screens/gh_user.dart#L336

Reply:
> BTW, I got the impression that the second argument of `bodyBuilder` here should work in this case:
> 
> https://github.com/git-touch/git-touch/blob/780970a5b754c7d7ceffb6f85a1b61383bbccf17/lib/screens/gh_user.dart#L336

Yeah, I saw this. setState was being called but the button was not refreshing. Anyways, I'll take a look at it again.

Reply:
I understood my mistake. I create a local variable inside `bodyBuilder` and `setState` was not reloading when I tried to update it. I had to update t.item4 directly. But since t.item4 is `final` I could only update it by creating a map of `isWatching` and `isStarred`. If I pass it in two separate boolean values, I cannot update them because they are final. Is this fix fine or is this there a better way to do it?

Reply:
Does this work?

```dart
tuple = tuple.withItem4('the new value')
```

Reply:
Oh, it seems won't work. The `tuple` is assigned to a new value and doesn't affect the original.

Reply:
This widget pattern should be improved indeed, maybe later

> creating a map of isWatching and isStarred

I think it's OK if we want to solve this temporarily. The other way is to declare a class and make its member not final:

```dart
class StatusPayload {
  bool isWatching;
  bool isStarred;
}
```

Reply:
Oh I should've used this method but it for some reason it's not working (button is not refreshing)

Reply:
Alright class seems better. I'll update it